openstack/objectstorage/v1/objects/results.go

```diff
@@ -66,14 +70,63 @@ func (r ObjectPage) IsEmpty() (bool, error) {
 
 // LastMarker returns the last object name in a ListResult.
 func (r ObjectPage) LastMarker() (string, error) {
-	names, err := ExtractNames(r)
+	return ExtractLastMarker(r)
```

Comment:
Since this is the only line in the `LastMarker` method, I debated not making a new function at all. Let me know if that's the better way to go.

Reply:
I think a separate function is OK if it'll be used by functions other than this one. Alternatively, if it'll only be used by this `LastMarker`, we could leave the function and lower-case it?

Reply:
Thanks for pointing this out. I saw a few ways of implementing this and wasn't sure the best method.

To confirm: you're saying leave everything as-is but rename `ExtractLastMarker` to `extractLastMarker`?

Is there any benefit to moving the body of the `ExtractLastMarker` function to the body of `LastMarker`?

Reply:
Oh nevermind - I read your comment too quickly. Separate function is fine :)

Reply:
Right, an unexported, separate function is OK.

> Is there any benefit to moving the body of the ExtractLastMarker function to the body of LastMarker?

If `ExtractLastMarker` is only called in one place (`LastMarker`), then we probably don't need it. Up to you.